src/software/thunderscope/common/proto_parameter_tree_util.py

```diff
@@ -2,6 +2,23 @@
 from pyqtgraph import parametertree
 from google.protobuf.json_format import MessageToDict
 from thefuzz import fuzz
+import netifaces
+
+
+"""
+Instead of the using the generic parameter parsing, this constant can be used to define custom handlers for specific
+fields in the proto.
+
+To define a custom handler:
+    1. The key is the name of the field in the proto
+    2. The value is the function that will be called to parse the field. Make sure the function is callable from this
+       file
+"""
+CUSTOM_PARAMETERS_OVERRIDE = {
+    "referee_interface": "__create_network_enum",
+    "robot_status_interface": "__create_network_enum",
+    "vision_interface": "__create_network_enum",
+}
```

Comment:
You can just store the function directly
```suggestion
CUSTOM_PARAMETERS_OVERRIDE = {
    "referee_interface": __create_network_enum,
    "robot_status_interface": __create_network_enum,
    "vision_interface": __create_network_enum,
}
```

Reply:
this is kind of a hack because python will complain about a `NameError`. so delaying the execution of this function using `eval` fixes it.

I could declare this constant at the bottom of the file or move it to a constants folder to not do this. I thought it'd be nice to declare this in the same file as the utility functions. I'm open to suggestions.

Reply:
I see, I thought python had function hoisting but I guess not